src/api_manager/context/request_context.cc

```diff
@@ -376,15 +378,18 @@ const std::string RequestContext::FindClientIPAddress() {
 void RequestContext::StartBackendSpanAndSetTraceContext() {
   backend_span_.reset(CreateSpan(cloud_trace_.get(), "Backend"));
 
-  // The span id in the header will be the backend span's id.
-  std::string trace_context = cloud_trace()->ToTraceContextHeader(
+  // TODO: A better logic would be to send for GRPC backends the grpc-trace-bin
+  // header, and for http/https backends the X-Cloud-Trace-Context header.
```

Comment:
This is a good idea.   Actually we have the data

It is here:  https://github.com/cloudendpoints/esp/blob/master/src/api_manager/context/request_context.cc#L296

Reply:
It should be an easy change.  Could you make this change in this PR

Reply:
I prefer to not do it in a rush right now, I would send you a different PR.